tools/cli/internal/cli/changelog/metadata/create_test.go

```diff
@@ -47,6 +47,18 @@ func TestCreateBuild_PreRun_InvalidVersion(t *testing.T) {
 	require.ErrorContains(t, opts.PreRun(), "invalid version date")
 }
 
+func TestCreateBuild_PreRun_UpcomingVersion(t *testing.T) {
+	fs := afero.NewMemMapFs()
+	opts := &Opts{
+		specRevision: "test",
+		runDate:      "2024-01-01",
+		versions:     []string{"2024-01-01.upcoming", "2025-01-01"},
```

Comment:
Q: is there chance for us to migrate from string to object with multiple keys in the future? 
 Not sure how much involved it would be, asking curious question.

 Thinking of all code generation tools we need to invent some parser for custom date format and prefixes